control/control.go

```diff
@@ -1016,6 +1016,54 @@ func (p *pluginControl) CollectMetrics(id string, allTags map[string]map[string]
 	return
 }
 
+func (p *pluginControl) StreamMetrics(
+	id string,
+	allTags map[string]map[string]string,
+	maxCollectDuration time.Duration,
+	maxMetricsBuffer int64) (chan []core.Metric, chan error, []error) {
+	if !p.Started {
+		return nil, nil, []error{ErrControllerNotStarted}
+	}
+	errs := make([]error, 0)
+	pluginToMetricMap, serrs, err := p.subscriptionGroups.Get(id)
+	if err != nil {
+		controlLogger.WithFields(log.Fields{
+			"_block":                "StreamMetrics",
+			"subscription-group-id": id,
+		}).Error(err)
+		errs = append(errs, err)
+		return nil, nil, errs
+	}
+
+	if serrs != nil {
```

Comment:
- errs is a slice of errors where we return any errors that happen during this call.
- err is the critical error coming from the call to subscriptionGroups.Get() (which is that this subscription doesn't exist). It is a stop this call error b/c if the subscription group doesn't exist we have nothing else we can do.
- serrs is coming from [this](https://github.com/intelsdi-x/snap/blob/master/control/subscription_group.go#L75-L78) and represents past errors with this subscription. I'll have to think about if maybe it doesn't make sense to include these? What do you think? cc: @jcooklin.